File: packages/typia/src/transformers/features/protobuf/ProtobufIsDecodeTransformer.ts
import { ProtobufIsDecodeProgrammer } from "../../../programmers/protobuf/ProtobufIsDecodeProgrammer";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufIsDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.isDecode",
      write: ProtobufIsDecodeProgrammer.write,
    });
}

File: packages/typia/native/transform/features/protobuf/ProtobufIsDecodeTransformer.go
package protobuf

import (
  shimast "github.com/microsoft/typescript-go/shim/ast"
  nativecontext "github.com/samchon/typia/packages/typia/native/core/context"
  nativeprotobufprogrammers "github.com/samchon/typia/packages/typia/native/core/programmers/protobuf"
  nativeinternal "github.com/samchon/typia/packages/typia/native/transform/internal"
)

type protobufIsDecodeTransformerNamespace struct{}

var ProtobufIsDecodeTransformer = protobufIsDecodeTransformerNamespace{}

func (protobufIsDecodeTransformerNamespace) Transform(props nativeinternal.ITransformProps) *shimast.Node {
  return nativeinternal.GenericTransformer.Scalar(nativeinternal.GenericTransformer_IProps{
    ITransformProps: props,
    Method:          "protobuf.isDecode",
    Write: func(x nativecontext.IProgrammerProps) *shimast.Node {
      return nativeprotobufprogrammers.ProtobufIsDecodeProgrammer.Write(nativeprotobufprogrammers.ProtobufIsDecodeProgrammer_IProps(x))
    },
  })
}
